src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3042,11 +3042,13 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
                                ? YMM_REGSIZE_BYTES
                                : XMM_REGSIZE_BYTES;
 
+        bool zeroing = false;
```

Comment:
We should be able to do this for any integral const that "fits in byte" or is the same bit pattern repeated to all bytes. So this should also work for `-1` for example which initializes all bytes to `0xFF`

Even if we don't handle that in this PR, a `TODO` and naming this something slightly different to indicate it applies to any case where the bytes are idempotent would be good.

Reply:
Right, but this path almost never hit for non-zero value so I didn't complicate the logic with "is input idempotent". Can leave a comment